libbeat/asset/asset.go

```diff
@@ -65,3 +70,61 @@ type Data struct {
 	Data     string
 	Package  string
 }
+
+func (d *Data) SetName(name string) {
+	d.Name = goTypeName(name)
+}
+
+func CreateAsset(license string, beat string, name string, pkg string, data []byte, priority string) ([]byte, error) {
+
+	// Depending on OS or tools configuration, files can contain carriages (\r),
+	// what leads to different results, remove them before encoding.
+	encData, err := EncodeData(strings.Replace(string(data), "\r", "", -1))
+	if err != nil {
+		return nil, errors.Wrap(err, "error encoding the data")
+	}
+
+	name = goTypeName(name)
+
+	var buf bytes.Buffer
+	Template.Execute(&buf, Data{
+		License:  license,
+		Beat:     beat,
+		Name:     name,
+		Data:     encData,
+		Priority: priority,
+		Package:  pkg,
+	})
+
+	bs, err := format.Source(buf.Bytes())
+	if err != nil {
+		return nil, errors.Wrap(err, "error creating golang file from template")
+	}
+
+	return bs, nil
+
+}
+
+// goTypeName removes special characters ('_', '.', '@') and returns a
+// camel-cased name.
```

Comment:
Why using camel cased names here?

Reply:
@ruflin This might not be necessary in this PR because it's applied to a different value than before. IIRC previously this was being used to camel case the generated Go function name like `AssetFileIntegrity()` rather than `AssetFile_integrity()`. And now it appears to be mutating the source file name that's referenced in the generated code.

Reply:
@andrewkroh It will be the case in a follow up PR. I just copied it over from there and is the reason it does not look optimal here. The more important point of this PR is bringing it into one place so will make it look good for this PR and can then still adjust it in the follow up PR.